rollup/circuitcapacitychecker/impl.go

```diff
@@ -219,3 +203,23 @@ func (ccc *CircuitCapacityChecker) SetLightMode(lightMode bool) error {
 
 	return nil
 }
+
+func MakeRustTrace(trace *types.BlockTrace, buffer *bytes.Buffer) unsafe.Pointer {
+	if buffer == nil {
+		buffer = new(bytes.Buffer)
+	}
+	buffer.Reset()
+
+	err := json.NewEncoder(buffer).Encode(trace)
+	if err != nil {
+		log.Error("fail to json marshal traces in MakeRustTrace", "err", err)
+		return nil
+	}
+
+	tracesStr := C.CString(string(buffer.Bytes()))
+	defer func() {
```

Comment:
It looks like we'll free the string before actually using it, should move this outside.

Reply:
how do you mean? L224 will have parse it by the time this deferred function is executed.

Reply:
Yes you're right, after parsing (which happens before returning from this function) the C string can be safely freed.